pkg/storage/sqlstorage/register.go

```diff
@@ -0,0 +1,33 @@
+package sqlstorage
+
+import (
+	"context"
+	"database/sql"
+	"runtime"
+	"strings"
+)
+
+var registeredGoMigrations []Migration
+
+type MigrationFunc func(ctx context.Context, schema Schema, tx *sql.Tx) error
+
+func RegisterGoMigration(fn MigrationFunc) {
+	_, filename, _, _ := runtime.Caller(1)
+	RegisterGoMigrationFromFilename(filename, fn)
+}
+
+func RegisterGoMigrationFromFilename(filename string, fn MigrationFunc) {
+	pathParts := strings.Split(filename, "/")
+	goFile := pathParts[len(pathParts)-1]
+	directory := pathParts[len(pathParts)-2]
+	number, name := extractMigrationInformation(directory)
+	engine := strings.Split(goFile, ".")[0]
+
```

Comment:
i feel like maybe we should check some of those values to be sure we have correct data maybe ?

Reply:
It seems to work, but maybe we could use the dedicated go package for that? https://pkg.go.dev/path/filepath#example-Split